pkg/apis/v1alpha1/cluster_run_template.go

```diff
@@ -45,7 +45,18 @@ type RunTemplateSpec struct {
 	// be created in the owner namespace. If the namespace is specified and is not
 	// the owner namespace, the resource will fail to be created.
 	// +kubebuilder:pruning:PreserveUnknownFields
-	Template runtime.RawExtension `json:"template"`
+	Template runtime.RawExtension `json:"template,omitempty"`
```

Comment:
we should also make this field a pointer instead of plain `rawextension` struct

e.g. given a runtemplate like

```yaml
---
apiVersion: carto.run/v1alpha1
kind: ClusterRunTemplate
metadata:
  name: sample
spec:
  ytt: |-
    #@ load("@ytt:data", "data")
    #@ load("@ytt:yaml", "yaml")
    apiVersion: v1
    kind: ConfigMap
    metadata:
      generateName: sample-
    data:
      context: foo
```

we end up hitting

```bash
> [hardcoded-breakpoint] github.com/vmware-tanzu/cartographer/pkg/templates.(*Stamper).Stamp() ./pkg/templates/stamper.go:148 (hits total:0) (PC: 0x1ed20e7)
   143:         var err error
   144:
   145:         runtime.Breakpoint()
   146:
   147:         switch {
=> 148:         case resourceTemplate.Template != nil:
   149:                 stampedObject, err = s.applyTemplate(resourceTemplate.Template.Raw)
   150:         case resourceTemplate.Ytt != "":
   151:                 stampedObject, err = s.applyYtt(ctx, resourceTemplate.Ytt)
   152:         default:
   153:                 err = fmt.Errorf("unknown resource template type, expected either template or ytt")


(dlv) p resourceTemplate
github.com/vmware-tanzu/cartographer/pkg/apis/v1alpha1.TemplateSpec {
        Template: *k8s.io/apimachinery/pkg/runtime.RawExtension {
                Raw: []uint8 len: 0, cap: 0, nil,
                Object: k8s.io/apimachinery/pkg/runtime.Object nil,},
        Ytt: "#@ load(\"@ytt:data\", \"data\")\n#@ load(\"@ytt:yaml\", \"yaml\")\napiVer...+79 more",
        Params: github.com/vmware-tanzu/cartographer/pkg/apis/v1alpha1.TemplateParams len: 0, cap: 0, nil,
        HealthRule: *github.com/vmware-tanzu/cartographer/pkg/apis/v1alpha1.HealthRule nil,}

```

which is not what we'd want (as we'd expect the `template` to evaluate to nil rather than the zero value of rawextension)

Reply:
```suggestion
	Template *runtime.RawExtension `json:"template,omitempty"`
```

Reply:
perhaps a better option is actually embedding `TemplateSpec` like we do for others?

e.g., for `ClusterImageTemplate` we do

```go
type ImageTemplateSpec struct {
        TemplateSpec `json:",inline"`
        ImagePath string `json:"imagePath"`
}
```

Reply:
(nvm, we have `healthRule` in it)